devicetypes/smartthings/zigbee-window-shade.src/zigbee-window-shade.groovy

```diff
@@ -0,0 +1,138 @@
+            state "level", action:"switch level.setLevel"
+        }
+
+        main "windowShade"
+        details(["windowShade", "contPause", "shadeLevel", "levelSliderControl", "refresh"])
+    }
+}
+
+private getCLUSTER_WINDOW_COVERING() { 0x0102 }
+private getATTRIBUTE_POSITION_LIFT() { 0x0008 }
+
+private List<Map> collectAttributes(Map descMap) {
+	List<Map> descMaps = new ArrayList<Map>()
+
+	descMaps.add(descMap)
+
+	if (descMap.additionalAttrs) {
+		descMaps.addAll(descMap.additionalAttrs)
+	}
+
+	return  descMaps
+}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+    log.debug "description:- ${description}"
+    if (description?.startsWith("read attr -")) {
+        Map descMap = zigbee.parseDescriptionAsMap(description)
+        if (descMap?.clusterInt == CLUSTER_WINDOW_COVERING && descMap.value) {
+            log.debug "attr: ${descMap?.attrInt}, value: ${descMap?.value}, decValue: ${Integer.parseInt(descMap.value, 16)}, ${device.getDataValue("model")}"
+            List<Map> descMaps = collectAttributes(descMap)
+            def liftmap = descMaps.find { it.attrInt == ATTRIBUTE_POSITION_LIFT }
+            if (liftmap) {
+                 if (liftmap.value == "64") { //open
+                    sendEvent(name: "windowShade", value: "open")
+                    sendEvent(name: "level", value: "100")
+                } else if (liftmap.value == "00") { //closed
+                    sendEvent(name: "windowShade", value: "closed")
+                    sendEvent(name: "level", value: "0")
+                } else {
+                    sendEvent(name: "windowShade", value: "partially open")
+                    sendEvent(name: "level", value: zigbee.convertHexToInt(attr.value))
+                }
+            }
+        }
+    }
+}
+
+def close() {
+    log.info "close()"
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x01)
+}
+
+def open() {
+    log.info "open()"
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x00)
+}
+
+def setLevel(data) {
+    log.info "setLevel()"
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x05, zigbee.convertToHexString(data, 2))
+}
+
+def pause() {
+    log.info "pause()"
+    zigbee.command(CLUSTER_WINDOW_COVERING, 0x02)
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+    return refresh()
+}
+
+def refresh() {
+    log.info "refresh()"
+    def cmds = zigbee.readAttribute(CLUSTER_WINDOW_COVERING, ATTRIBUTE_POSITION_LIFT)
+    return cmds
+}
+
+def configure() {
+    // Device-Watch allows 2 check-in misses from device + ping (plus 2 min lag time)
+    log.info "configure()"
+    sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+    log.debug "Configuring Reporting and Bindings."
+    zigbee.configureReporting(CLUSTER_WINDOW_COVERING, ATTRIBUTE_POSITION_LIFT, DataType.UINT8, 0, 600, null)
```

Comment:
Do we want a min reporting interval this low? I could see this behaving badly. Also I'm not sure how `null` has minReportableChange will work.

Reply:
0 means it will be reported whenever it changes subject to limitations of the device. I could see changing it to 1 which would mean report when it changes but no more often than 1/second. `null` isn't valid here